lib/friends/introvert.rb

```diff
@@ -429,16 +462,18 @@ def regex_friend_map
     #
     # The returned hash uses the following format:
     #   {
-    #     /regex/ => [list of friends matching regex]
+    #     /regex/ => [list of locations matching regex]
     #   }
     #
     # This hash is sorted (because Ruby's hashes are ordered) by decreasing
     # regex key length, so the key /Paris, France/ appears before /Paris/.
     #
     # @return [Hash{Regexp => Array<Friends::Location>}]
     def regex_location_map
-      @locations.each_with_object({}) do |location, hash|
```

Comment:
If we're only ever going to match one location and not be smart about it, can we keep this as a straight map?

Reply:
I'm not quite sure if i understand you here correct. We don't need something complex here, this is where I can follow you and agree.

So do you want something like before my changes, but that would also be a hash map.

Or only create a big map containing all names and alias. So something like
```ruby
def regex_location_map
  @locations.map { |location| [location.name, location.regex_for_name] }
end
```
Resulting in an `map[location_name, regex (with name and aliases)]`.

Reply:
Sorry for being unclear! I was picturing something like before your changes (and I realize that my `# @return` comment was also wrong 😬 ), basically giving us something like `{ location_name => regex }`. You're right that the way we use this it doesn't actually need to be a hash and could just be an array (which `map` returns) since we're just iterating through them all, but I slightly prefer the feeling of the hash to show how the data's related.

Does that make sense?

Reply:
No problem maybe i only understand it not correct.  Hope that i did now. I changed it back to a similar hash like before with `{location_name => Regexlist}`. It make sense yes. 

Also i changed the comment. Should now match more to the method.

Happens to the best 😃 .